File: Updog.Application/Comment/UseCases/Update/CommentUpdateParams.cs
using Updog.Domain;

namespace Updog.Application {
    /// <summary>
    /// Parameters to update a comment.
    /// </summary>
    public sealed class CommentUpdateParams {
        #region Properties
        public User User { get; }

        public int CommentId { get; }

        public string Body { get; }
        #endregion

        #region Constructor(s)
        public CommentUpdateParams(User user, int commentId, string body) {
            User = user;
            CommentId = commentId;
            Body = body;
        }
        #endregion
    }
}

File: Updog.Client/src/comment/use-cases/update/comment-update-params.ts
/**
 * Parameters to update a comment.
 */
export class CommentUpdateParams {
    /**
     * Create a new set of comment update parameters.
     * @param commentId The ID of the comment to update.
     * @param body The new body of the comment.
     */
    constructor(public commentId: number, public body: string) {}
}
